Add/devise

kennankole opened the pull request:
### Adding Devise to the app 
> ### Added;
* User registration 
* User login
* User email confirmation 
* User password reset
* Styles for user registration page and login pages

adebsa2401 committed:
Update README.md

Davidosky007 reviewed (commented):
Hi @kennankole @adebsa2401 👋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have the right project requirements you submitted a PR with `blog app add devise` changes for `Blog app: add authorization rules` Kindly submit this PR for the right project 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

arslanbisharat commented on app/models/user.rb:
```diff
@@ -1,7 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
- [ ] Please remove commented code as it makes code hard to read.

arslanbisharat commented on app/models/user.rb:
```diff
@@ -1,7 +1,14 @@
-  validates :post_counter, presence: false, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :email, presence: true
+  validates :photo, presence: false
+  validates :post_counter, presence: false
+  # numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
- [ ]  Please remove commented code as it makes code hard to read.

AmaduKamara replied:
- [ ] It would be nice if you could remove all commented unused code from your codebase as that makes your codebase not clean and professional. Let's try to have a clean and professional codebase and by so doing, we remove all commented unused codes from our codebase 👍🏽

arslanbisharat reviewed (changes requested):
Hi @kennankole, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AmaduKamara commented on app/controllers/application_controller.rb:
```diff
@@ -1,7 +1,14 @@
 class ApplicationController < ActionController::Base
-  helper_method :current_user
+  before_action :authenticate_user!, :configure_permitted_parameters, if: :devise_controller?
+  # before_action :store_user_location!, if: :storable_location?
```
- [ ] It would be nice if you could remove all commented unused code from your codebase as that makes your codebase not clean and professional. Let's try to have a clean and professional codebase and by so doing, we remove all commented unused codes from our codebase 👍🏽

AmaduKamara commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -0,0 +1,38 @@
+
+<%= stylesheet_link_tag 'registration' %>
+<section class="user-registration">
+<div class="reg-container">
+  <h3 class="reg-header">Sign up</h3>
+  <div class="sign-up-form">
+    <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+      <%= render "devise/shared/error_messages", resource: resource %>
+      <div class="field">
+        <%= f.label :name %><br />
+        <%= f.text_field :name, autofocus: true, autocomplete: "email" %>
+      </div>
+      <div class="field">
+        <%= f.label :email %><br />
+        <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+      </div>
+
+      <div class="field">
+        <%= f.label :password %>
+        <% if @minimum_password_length %>
+        <em>(<%= @minimum_password_length %> characters minimum)</em>
+        <% end %><br />
+        <%= f.password_field :password, autocomplete: "new-password" %>
+      </div>
+
+      <div class="field">
+        <%= f.label :password_confirmation %><br />
+        <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+      </div>
+
+      <div class="actions">
+        <%= f.submit "Sign up" %>
+      </div>
+    <% end %>
+  </div>
+  <%= render "devise/shared/links" %>
+  </div>
+</section>
```
Good job on styling the login and registration pages, but:

- [ ] When I try to register a new user, the app crashes and gives me the errors shown in the screenshot below. Kindly work on making registration work as expected and then the registered user is able to log in and view their profile and more...

<img width="1789" alt="Screenshot 2023-03-10 at 8 16 06 AM" src="https://user-images.githubusercontent.com/50941074/224262644-acdd7a60-5850-49ce-9c24-ba12b168296c.png">

AmaduKamara reviewed (changes requested):
# Status: Required Changes ♻️ ♻️ ♻️

Hi @kennankole,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Good commit messages 👍🏽
✅ Added styles for Login and registration pages 👍🏽
✅ Used devise as required 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Could you work on the registration functionality and all subsequent features to work as expected? Currently, I can't test other features as they are only assessed once a user in registered. Please fix them 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

kennankole committed:
removed all comments

kennankole committed:
Remove more comments

kennankole committed:
Fixed sending user email confirmation link in the routes
changed from
devise_for :users, controllers: { confirmations: 'users/confirmations' }
to
devise_for :users, controllers: { confirmations: 'confirmations' }

to solve for the following error
Routing Error
uninitialized constant Users when using email confirmation in devise

kennankole committed:
Fixing linter errors

omarsalem7 reviewed (changes requested):
Hi @kennankole 😊, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

![Almost near](https://media.giphy.com/media/ReyD4OdRiHeLDCRaEP/giphy.gif)

 ## Highlights
✔️ Use devise for authentication
✔️ Correct git-flow

 ## Required Changes ♻️
- [ ] As stated by the previous reviewer, kindly make sure to fix the error to make your app work properly and anyone using it can register without giving an error

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

kennankole committed:
Modified the _user_details.html.erb file

kennankole committed:
Merge branch 'development' into add/devise

rotimiazeez reviewed (approved):
Hi @kennankole 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the suggested changes from the previous CRs 👍🏽
- Good UI 👍🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
